### DISCUSSION TITLE
Support multiple content blocks using Nunjucks while honoring the data cascade

### DISCUSSION THREAD

**Discussion by euro** [11ty/eleventy#4119] [Enhancement Queue]
### Is your feature request related to a problem? Please describe.

The njk system supports extends and blocks but this seems to break Eleventy’s data cascade. 

### Describe the solution you'd like

I would like to see support for multiple named content blocks that can be implemented in templates and identified in the content files.

**Comment by zachleat** [comment]
Related in Liquid: https://liquidjs.com/tags/layout.html#Multiple-Blocks

Gray matter sections seems similar too: https://github.com/11ty/eleventy/pull/2502

**Comment by Seramis** [comment]
I'm missing this feature too - really would like to use this feature.

**Comment by euro** [comment]
I have a solution for this. I'll try to build a quick sample and link it up.

Link up this in eleventy.config:
```
import bundlerPlugin from '@11ty/eleventy-plugin-bundle';

export default async function (eleventyConfig) {
  eleventyConfig.addPlugin(bundlerPlugin);
};
```

uses this in the data file:
```
{% html "fooBar" %}
<p>Example.</p>
{% endhtml %}
```
and this in the template:

`{% getBundle "html", "fooBar" %}`

**Comment by Seramis** [comment]
@euro beware, nested bundles are not supported!
https://github.com/11ty/eleventy-plugin-bundle/issues/20

**Comment by zachleat** [comment]
Link to the Nunjucks docs: https://mozilla.github.io/nunjucks/templating.html#template-inheritance which also has a shout to https://jinja.palletsprojects.com/en/stable/templates/#template-inheritance

What exactly about the data cascade is missing here?
